File: backend/app/schemas/tag.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class TagCreate(BaseModel):
    name: str = Field(..., max_length=50)
    color: str = Field(default="#4f8ef7", max_length=7)


class TagUpdate(BaseModel):
    name: Optional[str] = Field(default=None, max_length=50)
    color: Optional[str] = Field(default=None, max_length=7)


class TagBrief(BaseModel):
    id: int
    name: str
    color: str

    model_config = {"from_attributes": True}


class TagResponse(TagBrief):
    task_count: int = 0
    created_at: datetime


class BulkTagRequest(BaseModel):
    task_ids: list[int] = Field(..., max_length=50)
    add_tag_ids: list[int] = []
    remove_tag_ids: list[int] = []


class BulkTagResponse(BaseModel):
    affected_tasks: int

File: frontend/src/types/tag.ts
export interface TagBrief {
  id: number;
  name: string;
  color: string;
}

export interface Tag extends TagBrief {
  task_count: number;
  created_at: string;
}

export interface CreateTagInput {
  name: string;
  color?: string;
}

export interface UpdateTagInput {
  name?: string;
  color?: string;
}

export interface BulkTagRequest {
  task_ids: number[];
  add_tag_ids?: number[];
  remove_tag_ids?: number[];
}

export interface BulkTagResponse {
  affected_tasks: number;
}

export const TAG_PRESET_COLORS = [
  { hex: "#4f8ef7", name: "Blue" },
  { hex: "#2dd4bf", name: "Teal" },
  { hex: "#a78bfa", name: "Violet" },
  { hex: "#fbbf24", name: "Amber" },
  { hex: "#f87171", name: "Red" },
  { hex: "#34d399", name: "Green" },
  { hex: "#f472b6", name: "Pink" },
  { hex: "#fb923c", name: "Orange" },
] as const;
